x-pack/plugins/global_search_bar/public/components/search_bar.tsx

```diff
@@ -293,11 +293,11 @@ export const SearchBar: FC<SearchBarProps> = (opts) => {
         console.log('Error trying to track searchbar metrics', err);
       }
 
-      if (event.shiftKey) {
+      if (event.shiftKey && isLoading === false) {
```

Comment:
Since `isLoading` gets updated in a debounced function, it doesn't have the right value fast enough for this bug fix to work, unfortunately :\ 

I'm not sure what the solution is. Maybe the debounced function needs to be split up, and the calls to `setIsLoading` need to not be debounced?

Reply:
I think adding the delay on the onChange is fixing the issue in my testing. What do you think?